### ISSUE TITLE
on="tap:" needs to enforce aria rules

### ISSUE THREAD

**Issue by cramforce** [ampproject/amphtml#340]
role=button and more importantly tabindex.

We need to warn about this ASAP.

**Comment by 88kbbq** [comment]
Does this mean this errors like MANDATORY_ATTR_MISSING A11Y: Element AMP-IMG#AMP_2 must have "role" attribute due to "tap" action, e.g. role="button" and MANDATORY_ATTR_MISSING A11Y: Element AMP-IMG#AMP_2 must have "tabindex" attribute due to "tap" action will go away? I can't figure out how to solve these validation errors, as these appear even after I add role and tabindex values.

**Comment by 88kbbq** [comment]
Oops! This is closed.

**Comment by dvoytenko** [comment]
Greg, could you please check if the validation issues persist?

**Comment by Gregable** [comment]
I tried this out on a local instance. A tag such as:

`<amp-img on="tap:" src="" width=10 height=10 role="" ta  bindex=""></amp-img>`
Produces the validation errors:
```
MANDATORY_ATTR_MISSING A11Y: Element AMP-IMG#AMP_1 must have "role" attribute due to "tap" action, e.g. role="button".
MANDATORY_ATTR_MISSING A11Y: Element AMP-IMG#AMP_1 must have "tabindex" attribute due to "tap" action.
```

While a tag such as:
`<amp-img on="tap:" src="" width=10 height=10 role="" tabindex=""></amp-img>`
Produces no validation errors.

@88kbbq, Are you still seeing these errors?

One small known issue is that these particular errors are not producing correct line numbers. The line/col will be 1:0 regardless of where the tags actually are.

**Comment by Gregable** [comment]
Added https://github.com/ampproject/amphtml/issues/1564 to track the line/col issue.

**Comment by 88kbbq** [comment]
The error line numbers are really what threw me off. I've added role and tabindex to all amp-img and the validation is successful. Never heard of those aria rules before. Thanks!
